include/NotificationsManager.php

```diff
@@ -827,4 +827,38 @@ private function formatIntros($intros) {
 
 		return $arr;
 	}
+
+	/**
+	 * @brief Check for missing contact data and try to fetch the data from 
+	 *     from other sources
+	 * 
+	 * @param array $arr The input array with the intro data
+	 * 
+	 * @return array The array with the intro data
+	 */
+	private function getMissingIntroData($arr) {
+		// If the network and the addr isn't available from the gcontact
+		// table entry, take the one of the contact table entry
+		if ($arr['gnetwork'] == "") {
+			$arr['gnetwork'] = $arr['network'];
+		}
+		if ($arr['gaddr'] == "") {
+			$arr['gaddr'] = $arr['addr'];
+		}
+
+		// If the network and addr is still not available
+		// get the missing data data from other sources
+		if ($arr['gnetwork'] == "" || $arr['gaddr'] == "") {
+			$ret = get_contact_details_by_url($arr['url']);
+		}
```

Comment:
Could you move this bracket to line 861?

I think the following two "if" constructs should only be called when "$ret" has got a value.